File: frontend/src/types/incident.ts
export type IncidentType =
  | "customer_issue"
  | "safety_concern"
  | "property_damage"
  | "ups_dispute"
  | "staff_incident"
  | "other";

export type IncidentSeverity = "low" | "medium" | "high";

export type IncidentStatus = "open" | "in_progress" | "resolved";

export type Incident = {
  id: number;
  title: string;
  incident_type: IncidentType;
  occurred_at: string;
  description: string;
  severity: IncidentSeverity;
  action_taken: string;
  follow_up_needed: boolean;
  follow_up_owner: string | null;
  follow_up_due_date: string | null;
  follow_up_notes: string | null;
  status: IncidentStatus;
  reported_by_user_id: number;
  created_at: string;
  updated_at: string;
};

export type IncidentAttachment = {
  id: number;
  incident_id: number;
  file_name: string;
  mime_type: string;
  file_size: number | null;
  storage_key: string;
  public_url: string | null;
  uploaded_by_user_id: number;
  created_at: string;
  view_url?: string | null;
  download_url?: string | null;
};

File: backend/app/schemas/incident.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime, date


INCIDENT_TYPES = {
    "customer_issue",
    "safety_concern",
    "property_damage",
    "ups_dispute",
    "staff_incident",
    "other",
}

SEVERITIES = {"low", "medium", "high"}

STATUSES = {"open", "in_progress", "resolved"}


class IncidentCreate(BaseModel):
    title: str
    incident_type: str
    occurred_at: datetime
    description: str
    severity: str
    action_taken: str
    follow_up_needed: bool = False
    follow_up_owner: Optional[str] = None
    follow_up_due_date: Optional[date] = None
    follow_up_notes: Optional[str] = None
    status: str = "open"


class IncidentUpdate(BaseModel):
    title: Optional[str] = None
    incident_type: Optional[str] = None
    occurred_at: Optional[datetime] = None
    description: Optional[str] = None
    severity: Optional[str] = None
    action_taken: Optional[str] = None
    follow_up_needed: Optional[bool] = None
    follow_up_owner: Optional[str] = None
    follow_up_due_date: Optional[date] = None
    follow_up_notes: Optional[str] = None
    status: Optional[str] = None


class IncidentOut(BaseModel):
    id: int
    title: str
    incident_type: str
    occurred_at: datetime
    description: str
    severity: str
    action_taken: str
    follow_up_needed: bool
    follow_up_owner: Optional[str]
    follow_up_due_date: Optional[date]
    follow_up_notes: Optional[str]
    status: str
    reported_by_user_id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class IncidentAttachmentOut(BaseModel):
    id: int
    incident_id: int
    file_name: str
    mime_type: str
    file_size: Optional[int]
    storage_key: str
    public_url: Optional[str]
    uploaded_by_user_id: int
    created_at: datetime
    view_url: Optional[str] = None
    download_url: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)
